src/main/java/io/r2dbc/postgresql/PostgresqlConnectionConfiguration.java

```diff
@@ -234,10 +235,13 @@ public Builder host(String host) {
          *
          * @param options the options
          * @return this {@link Builder}
-         * @throws IllegalArgumentException if {@code options} is {@code null}
+         * @throws IllegalArgumentException if {@code options} or any key or value of {@code options} is {@code null}
          */
         public Builder options(Map<String, String> options) {
-            this.options = Assert.requireNonNull(options, "options must not be null");
+            if (options == null) throw new IllegalArgumentException("options map must not be null");
+            if (options.keySet().stream().anyMatch(Objects::isNull)) throw new IllegalArgumentException("option keys must not be null");
```

Comment:
Note that `Stream` prohibits the use of `null` values. In general, we avoid using Java 8 streams because of these restrictions and performance issues. A `options.forEach((k, v) -> …)` seems more appropriate here.